service/user.go

```diff
@@ -0,0 +1,45 @@
+package service
+
+import (
+	"encoding/json"
+	"home-collect/domain"
+	"home-collect/repository"
+	"net/http"
+
+	"gopkg.in/mgo.v2/bson"
+)
+
+var userRepository = repository.UserRepository{}
+
+func init() {
+	userRepository.Connect()
+}
+
+func CreateUser(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+	var user domain.User
+
+	if err := json.NewDecoder(r.Body).Decode(&user); err != nil {
+		respondWithError(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	user.ID = bson.NewObjectId()
+	if err := userRepository.InsertUser(user); err != nil {
```

Comment:
@edvmorango if  not insert so operation fail and return error else all right

Reply:
Sure @adrissonGalvao .